docs/maintainer/infrastructure.md

```diff
@@ -469,6 +469,159 @@ uploaded from the new feedstock, by design.
 Once this is done correctly and the package is uploaded,
 you can then request the conda-forge core devs to archive the old feedstock.
 
+## How package building interacts with infrastructure
```

Comment:
This is also explaining the "life cycle" of the package. Maybe a different title is useful. Some suggestions:

- Life cycle of a package and the involved infrastructure
- The different stages of package building and the infrastructure used